src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -311,31 +314,24 @@ private void enterPosition(Spread spread) {
      *
      * @param exchange The exchange to fetch metadata from.
      * @param currencyPair The currency pair to look for.
+     * @param defaultScale If the scale cannot be found, this value will be returned.
      * @return The number of decimals allowed for a volume in this currency pair on this exchange.
      */
-    Integer computeVolumeScale(Exchange exchange, CurrencyPair currencyPair) {
-        final CurrencyPairMetaData defaultPairMetaData = new CurrencyPairMetaData(
-            BigDecimal.ZERO,
-            BigDecimal.ZERO,
-            BigDecimal.valueOf(Long.MAX_VALUE),
-            BTC_SCALE,
-            BTC_SCALE,
-            new FeeTier[0],
-            Currency.BTC
-        );
-
+    Integer computeVolumeScale(Exchange exchange, CurrencyPair currencyPair, Integer defaultScale) {
         final ExchangeMetaData exchangeMetaData = exchange.getExchangeMetaData();
-        final CurrencyPairMetaData currencyPairMetaData = exchangeMetaData.getCurrencyPairs().getOrDefault(
-            currencyPair,
-            defaultPairMetaData
-        );
-
-        if (currencyPairMetaData.getVolumeScale() == null) {
-            LOGGER.debug("Defaulting to scale of {} for volume because metadata is unavailable", BTC_SCALE);
-            return BTC_SCALE;
+        final Optional<CurrencyPairMetaData> currencyPairMetaDataOptional = Optional.ofNullable(exchangeMetaData.getCurrencyPairs().get(currencyPair));
+
+        if (currencyPairMetaDataOptional.isPresent()) {
+            CurrencyPairMetaData currencyPairMetaData = currencyPairMetaDataOptional.get();
+
+            if (currencyPairMetaData.getVolumeScale() != null) {
+                LOGGER.debug("{} {} volume scale found in metadata: {}", exchange.getExchangeSpecification().getExchangeName(), currencyPair, currencyPairMetaData.getPriceScale());
+                return currencyPairMetaData.getVolumeScale();
+            }
         }
 
-        return currencyPairMetaData.getVolumeScale();
+        LOGGER.warn("{} {} price scale unavailable from exchange metadata, using default: {}", exchange.getExchangeSpecification().getExchangeName(), currencyPair, defaultScale);
+        return defaultScale;
```

Comment:
What is the advantage of changing the scale from 8 (BTC_SCALE) to 1 (the defaultScale we pass in now)?

Are you aware this scale value will be used in `TradeVolume` and it's subclasses?

Reply:
Not all coins accept 8 decimal places when placing orders, but almost all of them should accept 1 decimal place. The idea was to find a default that is the most compatible across all exchanges and all coins, and sacrificing some precision for compatibility seems worthwhile.

In a future PR I'm going to add configuration that you can use if your coin is defaulting because XChange doesn't have metadata for it but you want something other than 1.